### ISSUE TITLE
TaintedSql sample from documentation no longer works with psalm 5.x

### ISSUE THREAD

**Issue by cgocast** [vimeo/psalm#9897]
When I run the sample from https://psalm.dev/docs/running_psalm/issues/TaintedSql/ with psalm 4.30.0, I get an "ERROR: TaintedSql" as expected.

With psalm 5.12.0, the expected error is missing. I get " No errors found!".

**Comment by orklah** [comment]
I get the error on latest master though:
https://psalm.dev/r/cc4f1a6781

Can you make sure your config is correctly set up?

**Comment by cgocast** [comment]
Thank you @orklah for checking on your side !

On my side, I am running PHP 8.2.6 on Linux with the latest master
```
$> git log
commit 49c6f3107192760a94f993f423f0e2b063b7c5a6 (HEAD -> master, origin/master, origin/HEAD)
Merge: 2c50745d8 51fd63cfb
Author: orklah <[orklah@users.noreply.github.com](mailto:orklah@users.noreply.github.com)>
Date:   Tue Jun 13 18:24:53 2023 +0200

    Merge pull request #9915 from kkmuffme/fix-psalm-6-warning-findUnusedCode
    
    fix CI warnings for findUnusedCode in psalm 6
```

Here is my configuration file
```
$> cat psalm.xml 
<?xml version="1.0"?>
<psalm
    errorLevel="4"
    resolveFromConfigFile="true"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
    findUnusedBaselineEntry="true"
>
    <projectFiles>
        <file name="tainted_sql.php" />
        <ignoreFiles>
            <directory name="vendor" />
        </ignoreFiles>
    </projectFiles>
</psalm>
```

I am still not able to get the **ERROR: TaintedSql**
```
$> php vendor/vimeo/psalm/psalm --taint-analysis tainted_sql.php 
Warning: "findUnusedCode" will default to "true" in Psalm 6. You should explicitly enable or disable this setting.

Install the opcache extension to make use of JIT on PHP 8.0+ for a 20%+ performance boost!

Target PHP version: 8.2 (inferred from current PHP version).
Scanning files...
Analyzing files...

░

To whom it may concern: Psalm cannot detect unused classes, methods and properties
when analyzing individual files and folders. Run on the full project to enable
complete unused code detection.
------------------------------
                              
       No errors found!       
                              
------------------------------

Checks took 0.57 seconds and used 72.122MB of memory
Psalm was able to infer types for 100% of the codebase
```

I also tried on Windows with PHP 8.2.6 and PHP 8.1.19 and I am still not able to get the **ERROR: TaintedSql** with psalm 5.12.0.


When I switch back to psalm 4.30.0 I am able to get the error both on Windows and Linux.


Please @orklah can you point me where I should look in order to get the error ?

**Comment by orklah** [comment]
Can you try the following things separately:
- run psalm with `php vendor/vimeo/psalm/psalm --taint-analysis` without the file after
- run psalm with `php vendor/vimeo/psalm/psalm` but enable this config in your config file: https://psalm.dev/docs/running_psalm/configuration/#runtaintanalysis

**Comment by cgocast** [comment]
With `php vendor/vimeo/psalm/psalm --taint-analysis`, I still get "No errors found!"

With `php vendor/vimeo/psalm/psalm` and `runTaintAnalysis="true"` in psalm.xml, I also get "No errors found!"

**Comment by orklah** [comment]
Oh, I think I get it. Can you add

```
<enableExtensions>
        <extension name="pdo"/>
    </enableExtensions>
```

in your config file?

**Comment by cgocast** [comment]
It worked ! I am now able to get the **ERROR: TaintedSql** both on Linux and Windows.

Thank you @orklah

**Comment by orklah** [comment]
It's great that we fixed the issue but I'm not very satisfied with the fact that Psalm knew the symbols but only partially

@AndrolGenhald IIRC, we planned to remove extension's symbols from the callmap, I'm not sure it was done for all extensiojs

**Comment by AndrolGenhald** [comment]
Yeah, the framework is there but it was never finished. Need to get more extension stubs added under `stubs/extensions` and removed from the callmap, but I haven't had any free time lately and it's not exactly at the top of my list when I do. Happy to help someone else figure it out though!
